persistence/src/main/java/com/epam/esm/dao/impl/CertificateDAO.java

```diff
@@ -0,0 +1,183 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.AbstractDAO;
+import com.epam.esm.dao.mapper.CertificateMapper;
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.search.SearchQueryBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+
+import java.math.BigInteger;
+import java.util.Collection;
+
+/**
+ * Class that interacts with the database
+ *
+ * @author Andrey Belik
+ * @version 1.0
+ */
+@Component
+public class CertificateDAO implements AbstractDAO<Certificate, BigInteger> {
+  private final JdbcTemplate jdbcTemplate;
+
+  private static final String SQL_FIND_ALL =
+      "SELECT id, name, description, price, duration,"
+          + " create_date, last_update_date FROM gifts.certificate";
+  private static final String SQL_FIND_ALL_BY_TAG_ID =
+      SQL_FIND_ALL
+          + " JOIN gifts.certificate_tag ON "
+          + "gifts.certificate.id = gifts.certificate_tag.certificate_id"
+          + " where gifts.certificate_tag.tag_id=?";
+  private static final String SQL_FIND_BY_ID = SQL_FIND_ALL + " WHERE id=?";
+  private static final String SQL_ADD =
+      "INSERT INTO gifts.certificate"
+          + "(name, description, price, duration, create_date, last_update_date)"
+          + " values(?, ?, ?, ?, ?, ?)";
+  private static final String SQL_UPDATE =
+      "UPDATE gifts.certificate SET "
+          + "name=?, description=?, price=?, duration=?, create_date=?,"
+          + " last_update_date=? where id=?";
+  private static final String SQL_DELETE = "DELETE FROM gifts.certificate WHERE id=?";
+  private static final String SQL_FIND_ALL_BY_TAG_NAME =
+      "SELECT certificate.id, certificate.name, description, price, duration,"
+          + " create_date, last_update_date FROM gifts.certificate"
+          + " JOIN gifts.certificate_tag ON gifts.certificate.id = gifts.certificate_tag.certificate_id"
+          + " JOIN gifts.tag ON gifts.certificate_tag.tag_id = gifts.tag.id WHERE tag.name = ?";
+
+  /**
+   * Constructor
+   *
+   * @param jdbcTemplate {@link org.springframework.jdbc.core.JdbcTemplate}
+   */
+  @Autowired
+  public CertificateDAO(JdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  /**
+   * Find all tags method method
+   *
+   * @return {@link java.util.Collection} of certificates
+   */
+  @Override
+  public Collection<Certificate> findAll() {
+    return jdbcTemplate.query(SQL_FIND_ALL, new CertificateMapper());
```

Comment:
@belikandrey Could you inject the mapper using Spring features instead of constantly creating it yourself?